src/R/Support/Impl/Help/Packages/PackageIndex.cs

```diff
@@ -173,5 +187,22 @@ public sealed class PackageIndex : IPackageIndex {
                 Directory.Delete(CacheFolderPath, recursive: true);
             } catch (IOException) { } catch (UnauthorizedAccessException) { }
         }
+
+        private void ScheduleIdleTimeRebuild() {
+            IdleTimeAction.Cancel(typeof(PackageIndex));
+            IdleTimeAction.Create(() => RebuildIndex(), 100, typeof(PackageIndex), _shell);
+        }
+
+        private void RebuildIndex() {
+            if(!_buildIndexLock.IsCompleted) {
```

Comment:
There is a method `_buildIndexLock.ResetIfNotWaiting`, which returns `true` if reset succeeded and false if we are waiting:

```{csharp}
if (_buildIndexLock.ResetIfNotWaiting()) {
    _packages.Clear();
    BuildIndexAsync().DoNotWait();
} else {
    // Stull building, try again later
    ScheduleIdleTimeRebuild();
}
```